tests/integration_tests/bugs/test_lp1900837.py

```diff
@@ -0,0 +1,28 @@
+"""Integration test for LP: #1900836.
+
+This test mirrors the reproducing steps from the reported bug: it changes the
+permissions on cloud-init.log to 600 and confirms that they remain 600 after a
+reboot.
+"""
+import pytest
+
+
+def _get_log_perms(client):
+    return client.execute("stat -c %a /var/log/cloud-init.log")
+
+
+@pytest.mark.sru_2020_11
+class TestLp1900836:
```

Comment:
I don't think we should name files/classes/tests after bugs. The docstring at the top has the bug number, and we could perhaps add a comment with the bug number somewhere else, but I'd like to be able to quickly glance a file/class/test name to know what the test is doing.

Reply:
I'm wondering if we are abusing marks here as a generic tagging feature. If we adopt a class or test naming convention that includes SRU or SRU_20_4 (cloud-init Major_minor version or date-based SRU start 2020_11)  wouldn't be be able to provide pytest keyword search expressions any avoid the bload of marker definitions in tox? they we could run things like pytest -k "SRU" to run any SRU-related tests ever defined or -k "SRU_2020_11"  to match only specific tests related to SRU_2020_11?

Reply:
> I don't think we should name files/classes/tests after bugs. The docstring at the top has the bug number, and we could perhaps add a comment with the bug number somewhere else, but I'd like to be able to quickly glance a file/class/test name to know what the test is doing.

I agree I'd like to glance at test name and know what it's doing, but I also want a way to search and run specific known tests based on my intent:
  - am I SRU testing and can I easily select the subset of tests related to that capacity with a -k or -m param?
  - am I trying to validate just one bug and want to provide a bug Id.

 This is probably just representing something might be able to be solved per @TheRealFalcon's generic kwargs handling suggestion above or maybe we create a decorator that could optionally skip test or not include a test that didn't match a set of kwargs or environment variable criteria.

Reply:
> I don't think we should name files/classes/tests after bugs. The docstring at the top has the bug number, and we could perhaps add a comment with the bug number somewhere else, but I'd like to be able to quickly glance a file/class/test name to know what the test is doing.

That's a good point; I think we probably do want the bug number in at least one part of the test ID (so `-k 1900837` works, for example, but also so you _can_ easily find the bug based on test output). I'll play around with some naming combinations and see what I like the look of.

> I'm wondering if we are abusing marks here as a generic tagging feature (which I guess it actually is :) ).

We are using it that way, and I don't think it's an abuse. :p

> If we adopt a class or test naming convention that includes SRU or SRU_20_4 (cloud-init Major_minor version or date-based SRU start 2020_11)

Using cloud-init versions will fall over as soon as we have a non-upstream-release SRU (e.g. `20.2-45-g5f7825e2-0ubuntu1` was our last SRU); that's hard to fit into a mark name, it's even harder to fit into a test name. That's why I went with date-based. (If we're ever unfortunate enough to start two SRUs in a month, we can do `sru_2020_11a` or similar.)

> wouldn't we be able to provide pytest keyword search expressions any avoid the bloat of marker definitions in tox? they we could run things like `pytest -k "SRU"` to run any SRU-related tests ever defined or `pytest -k "SRU_2020_11"` to match only specific tests related to SRU_2020_11?

The advantage marks bring is that we gain validation of them: if we misname a test method/class "SRU_2200_11" and it makes it through code review, then we may never fix that (and so may not run the test as part of SRU validation). Using `@pytest.mark.sru_2200_11` errors out without running any tests:

```
========================================== ERRORS ==========================================
_____________ ERROR collecting tests/integration_tests/bugs/test_lp1900837.py ______________
'sru_2200_11' not found in `markers` configuration option
================================= short test summary info ==================================
ERROR tests/integration_tests/bugs/test_lp1900837.py
!!!!!!!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!!!!!!!
===================================== 1 error in 0.24s =====================================
```

> > I don't think we should name files/classes/tests after bugs. The docstring at the top has the bug number, and we could perhaps add a comment with the bug number somewhere else, but I'd like to be able to quickly glance a file/class/test name to know what the test is doing.
> 
> I agree I'd like to glance at test name and know what it's doing, but I also want a way to search and run specific known tests based on my intent:
> 
>     * am I SRU testing and can I easily select the subset of tests related to that capacity with a -k or -m param?
> 
>     * am I trying to validate just one bug and want to provide a bug Id.
> 
> 
> This is probably just representing something might be able to be solved per @TheRealFalcon's generic kwargs handling suggestion above or maybe we create a decorator that could optionally skip test or not include a test that didn't match a set of kwargs or environment variable criteria.

We shouldn't need to write any custom code for this: some combination of marking, test naming and `-k`/`-m` should be sufficient. I think, basically, we'll want to declare that bug number must be included in some part of the pytest nodeid, so that `-k` selection can work, and use marks for tagging tests as SRU-relevant. For your first case, `-m sru_2020_11` will select this SRU's tests, and for your second case `-k 1900837` will select the bug's tests.

(I can't think of a particular reason we would want to run "all SRU tests" instead of just "all tests", but `-k` does select based on substrings of mark names, so `-k sru` would get you that.)